com.unity.sg2/Editor/GraphDeltaRegistry/Registry/SamplerType.cs

```diff
@@ -0,0 +1,172 @@
+        public RegistryFlags GetRegistryFlags() => RegistryFlags.Type;
+
+        #region LocalNames
+        public const string kFilter = "Filter";
+        public const string kWrap = "Wrap";
+        public const string kAniso = "Aniso";
+        public const string kCompare = "Compare";
+        #endregion
+
+        private static string GetUniqueSamplerName(FieldHandler field)
+        {
+            return field.ID.FullPath.Replace(".", "_");
+        }
+
+        public enum Filter { Point, Linear, Trilinear }
+        public enum Wrap { Clamp, Repeat, Mirror, MirrorOnce } // optionally can be per component- can be added later.
+        public enum Aniso { None = 0, Ansio2 = 2, Ansio8 = 8, Ansio16 = 16 } // optional
+
+        #region GetSet
+        public static void SetDepthComparison(FieldHandler field, bool enable)
+        {
+            try     { field.GetSubField<bool>(kCompare).SetData(enable); }
+            catch   { field.AddSubField<bool>(kCompare, enable); }
+        }
+        public static bool GetDepthComparison(FieldHandler field)
+        {
+            try     { return field.GetSubField<bool>(kCompare).GetData(); }
+            catch   { return false;  }
+        }
+        public static void SetAniso(FieldHandler field, Aniso aniso)
+        {
+            try     { field.GetSubField<Aniso>(kAniso).SetData(aniso); }
+            catch   { field.AddSubField<Aniso>(kAniso, aniso); }
+        }
+        public static Aniso GetAniso(FieldHandler field)
+        {
+            try     { return field.GetSubField<Aniso>(kAniso).GetData(); }
+            catch   { return Aniso.None; }
+
+        }
+        public static void SetFilter(FieldHandler field, Filter filter)
+        {
+            try     { field.GetSubField<Filter>(kFilter).SetData(filter); }
+            catch   { field.AddSubField<Filter>(kFilter, filter); }
+        }
+        public static Filter GetFilter(FieldHandler field)
+        {
+            try     { return field.GetSubField<Filter>(kFilter).GetData(); }
+            catch   { return Filter.Linear; }
+        }
+
+        public static void SetWrap(FieldHandler field, Wrap wrap)
+        {
+            try     { field.GetSubField<Wrap>(kWrap).SetData(wrap); }
+            catch   { field.AddSubField<Wrap>(kWrap, wrap); }
+        }
+        public static Wrap GetWrap(FieldHandler field)
+        {
+            try     { return field.GetSubField<Wrap>(kWrap).GetData(); }
+            catch   { return Wrap.Repeat; }
+        }
+
+        public static bool IsInitialized(FieldHandler field)
+        {
+            return field.GetSubField<Filter>(kFilter) != null
+                || field.GetSubField<Wrap>(kWrap) != null
+                || field.GetSubField<Aniso>(kAniso) != null
+                || field.GetSubField<bool>(kCompare) != null;
+        }
+
+
+        #endregion
+
+
+        public void BuildType(FieldHandler field, Registry registry)
+        {
```

Comment:
SamplerStates, in Unity, have a very specific set of behavior requirements. They are parsed as strings and aren't constructed or composed objects. For this reason, we can make a lot of assumptions about SamplerState initialization and avoid any concrete data generation.